.github/workflows/release-preview.yml

```diff
@@ -0,0 +1,60 @@
+name: release
+
+on:
+  pull_request_review:
+    types: [submitted]
+  workflow_dispatch:
+
+jobs:
+  check:
+    # Trigger the permissions check whenever someone approves a pull request.
+    # They must have the write permissions to the repo in order to
+    # trigger preview package publishing.
+    if: github.event.review.state == 'approved'
+    runs-on: ubuntu-latest
+    outputs:
```

Comment:
## Setup

The trick to a finally properly working setup was:

- Has 2 separate jobs: one checks for permissions, the other publishes the preview;
- The check job shares its outputs via `outputs`. It references the used action output based on the user's permissions. The job itself never fails because then the entire PR status will be failed, which isn't nice;
- The preview job `needs` the check job BUT also needs a custom `if` condition since the check job won't ever fail (even if it's skipped, GitHub still treats the job status as `success`). In the `if`, the preview job can check the shared `outputs` from the permissions check, and publish the preview only if the output is `'true'`.